### DISCUSSION TITLE
Airspeed sensor in the wing

### DISCUSSION THREAD

**Discussion by ViorelX** [iNavFlight/inav#10563] [Show and tell]
I have a Volantex Phoenix 2400 rc airplane prepared with FC SpeedyBee F405 Mini & INAV 7.1.2.
I want to add an airspeed sensor, but I don't like the regular Pitot tube: it's ugly and adds drag.

First of all, I have got one of the cheapest differential pressure sensors: XGZP6897 1kPa, both analog and digital variant. I didn't manage to make the digital variant work in the INAV system, thus I go further with the analog variant: XGZP6897A.
The sensor will be located inside the wing and I am taking into consideration where to bring its air ports:

**Solution 1**: The air ports of the sensor are on the upper and lower wing surface:

![senzor presiune aripa 1](https://github.com/user-attachments/assets/478de9b1-f599-4355-ae2e-6ff641e088b8)

During the flight there is a static pressure difference between the upper and lower wing surface: this generates lift. For a given angle of attack this pressure difference is proportional to the squared airspeed. Is the angle of attack larger, this pressure difference becomes greater. To prevent stall, what actually counts is not the speed itself, but the differential pressure.
For example, my plane weighs 2kg and its wings surface is about 0.45 m2. The required average differential pressure between the upper and lower surface should be about 60Pa. What the sensor this way measures is somewhat proportional to this average differential pressure. Such way, I will simply check if this differential pressure is large enough, regardless of the airspeed and angle of attack. Thus, I don't need to measure the airspeed and to take into consideration the angle of attack.

Is this really a great new idea, or it is and old one and not so great?
Let's analyze things with JavaFoil, a computer program which gives the pressure distribution over the airfoils:
[Cp2.pdf](https://github.com/user-attachments/files/18252370/Cp2.pdf)
I put there a wing profile close to the actual wings of my plane and a set of angles of attack: 0, 2, 4, 6 degrees. I got:
|AOA|  C_lift |   Delta Cp
|  0      |  0.54  |  0.9
|  2      |  0.77  |  1.3
|  4      |  0.99  |  1.6
|  6      |  1.16  |  2.0

Seemingly, it is OK: on increasing the angle of attack AOA, the lift coefficient also increases and the indicated pressure difference (proportional to Delta Cp for a given speed) also increases, in a relative proportional manner to the lift coefficient. Therefore, I can say this sensor directly measures the lift.
This is what I am actually looking for!
Notice, this can be valid only for a certain range of AOA (but within normally operating conditions).
However, I am not sure this is a good solution. The dependence between  D_P and C_lift is very sensitive to the wing profile, ports locations and is not always so simply linear.

**Solution 2**: One air port takes the stagnation pressure (i.e. the total pressure) from the leading edge, whereas the other air port is left free, to get the pressure from the inside of the wing, this being the static pressure (since there is a channel between this cavity inside the wing and the fuselage):

![senzor presiune aripa 2](https://github.com/user-attachments/assets/51bc16a1-17c8-4516-89ec-873ff4bad28d)

It seems this way it works like a regular Pitot tube, measuring the dynamic pressure.
However, JavaFoil shows there is not a precise location to get the stagnation pressure. It is very sensitive to the angle of attack.

**Solution 3**: Finally, I have to admit there is not any convenient alternative to the Pitot tube.

![senzor presiune aripa 3](https://github.com/user-attachments/assets/e5e4d763-501a-46f4-8c2b-83d5dd127521)

However, I will put there a very tiny tube over the wing profile to take the total pressure. The static pressure will be taken from inside the wing. It is ugly, but anyhow, the tube will be very thin compared to a regular Pitot tube, which includes the static pressure channel around the central tube of the total pressure.
Most likely, I will proceed with this last solution. The pressure taken from this tiny tube is in fact not very sensitive to AOA. I should admit this conventional, well established approach: one measures the airspeed and judges if it is great enough to ensure the required lift. Directly measuring of lift by method 1 is neither accurate nor really meaningful.

**Comment by sensei-hacker** [comment]
Your reasoning certainly makes sense to me. I thought about doing something similar for a stall sensor.

As you may know, nobody does it that, measuring the pressure at the wing surface - in models or full-size. This may perhaps be related to the fact that pressure at one particular place on the wing isn't representative of the average pressure across the wing. You may be familiar with how a stall progresses, the airflow separation starting in one area and spreading. One would need to very carefully choose the spot you want to Sensei the stall.

For sensing stalls specifically, there are two types of sensors commonly used. One is essentially a toggle switch mounted on the leading edge at the appropriate angle. The other uses differential pressure again on the leading edge to sense when the stagnation bubble passes the sensed point.

**Comment by ViorelX** [comment]
We can distinguish two types of stall, caused by:
- too low speed;
- too large angle of attack.

If the RC plane is stabilized and is operated normally we only should care for the speed. This is easier to handle, sensing the air near the leading edge.
Flow separation responsible for the second type of stall begins from the trailing edge and should be handled with sensors in this area.
I think it is satisfactory to limit to the first case: airplane operated with AOA within the normal range.

**Comment by sensei-hacker** [comment]
ALL stalls are ALWAYS AoA beyond the critical angle of attack, generally 15°-16.5°.

Airspeed is only useful when you know that in a particular configuration, while flying straight and level, you will exceed the critical AoA at a particular speed. That is, airspeed is a proxy for AoA. A proxy that is valid only in a specific configuration and orientation.

**Comment by ViorelX** [comment]
It took some time to understand this.
Indeed, let's assume you fly level and start reducing speed. On stall the plane will descend very steeply. Even if the attitude will be maintained (relative to the horizontal direction), the AoA increases, this being measured relative to the speed direction.

Now it seems to me it is not very practical to have a stall sensor, rather than a regular airspeed sensor. What I have analyzed as solution 1 measures the lift (with doubtful accuracy) and may indicate good lift even when you are close to stall. I think it is better to monitor the airspeed.
Let's say my plane has a stall speed of 7m/s (for a certain AoA) and I cruise normally with 13m/s. I take care for the airspeed and I know on 10-9-8 m/s I am still safe, at least for a proper AoA. A lift sensor may indicate about the same value of lift for a broad range of speed, down to the stall speed and you risk being too late to recover from stall.

In conclusion, I will adopt solution 3.

**Comment by sensei-hacker** [comment]
Using the speed is a good approach, used often in full sized. Just be aware that speed is only valid in straight and level flight - if you turn the necessary speed to avoid stall increases. The speed with flaps is different then without. Adding any weight, such as a larger battery, will change the speed.

The stall angle will always be the same  regardless of weight etc - 16°.   But AoA sensors are finicky.

Remember as you approach stall, the air ABOVE the wing is disturbed. You want your pitot in *undisturbed* air. (That is, not above the wing.)

For the pitot tube sensing the stagnation pressure, your tube will be shorter if it goes directly into the airstream in front, before the airstream is affected by the wing. Going above the wing as illustrated above will require a much longer tube to get out of the aerodynamic effects of the wing, especially as the wing nears stall.  You'd need to be in the area where the streamlines are straight, as seen in this image of a wing in a wind tunnel.

![IMG_20241228_153316](https://github.com/user-attachments/assets/d7bc6c8e-c9ad-46fb-8cab-29de5ff2958a)

**Comment by ViorelX** [comment]
Yes, you are right. It is wrong to place the air gauge above the wing. Not only because the air is strongly disturbed as you approach stall. Before stall, there is a too much variation of air speed with AoA.
Under the wing is the air stream slower and doesn't vary so much with AoA, but it is not much convenient, too.

Last days I was going to implement solution 3:

![preparare aripa](https://github.com/user-attachments/assets/0bf18552-a2b6-4728-be0c-f853b88871bd)

![aripa preparata 2](https://github.com/user-attachments/assets/4d46cbca-30e4-4102-b800-79140f42caf4)

After the work I changed my mind, being more and more convinced that what I have done is wrong. Then I proceeded to restore the wing:

![restaurare aripa](https://github.com/user-attachments/assets/4c2e7ceb-7824-4489-964c-691c97173dbb)

![aripa restaurata](https://github.com/user-attachments/assets/dd178505-4c5e-404b-9128-0703187a5a79)
No more hole on the upper side of the wing.

I have purchased 2 cheap sensor from Aliexpress to test them: XGZP6897A. According to the datasheet, they should output the analog signal in the range 0.5V-4.5V. Actually, one of them outputs in the range 0-1V, the other 0-0.45V.
What a stupidity! What a waste of time!
It will take a while until I will replace that shitty sensors and release this project.
